apps/builder/app/builder/features/settings-panel/use-settings-logic.ts

```diff
@@ -0,0 +1,44 @@
+import type { Instance } from "@webstudio-is/project-build";
+import store from "immerhin";
+import { type KeyboardEventHandler, useRef } from "react";
+import { instancesStore } from "~/shared/nano-states";
+
+type SettingUpdate = { setting: "label"; value: string };
+
+export const useSettingsLogic = ({
+  selectedInstance,
+}: {
+  selectedInstance: Instance;
+}) => {
+  const changes = useRef<Map<SettingUpdate["setting"], SettingUpdate["value"]>>(
+    new Map()
+  );
```

Comment:
Types won't work if one setting change value type. Let it be just an object. Map doesn't optimize here anything.